frontend/packages/console-shared/src/hooks/usePostFormSubmitAction.ts

```diff
@@ -0,0 +1,49 @@
+import { useCallback, useMemo } from 'react';
+import { INCONTEXT_ACTIONS_CONNECTS_TO } from '@console/dev-console/src/const';
+import { K8sResourceCommon } from '@console/internal/module/k8s';
+import { doConnectsToBinding } from '@console/topology/src/utils/connector-utils';
+import { useQueryParams } from './useQueryParams';
+
+/**
+ * Reads an action from the current query params and return a callback to
+ * connect a newly created resource to an existing one in the topology view
+ */
+export const usePostFormSubmitAction = <
+  R extends K8sResourceCommon[] | K8sResourceCommon = K8sResourceCommon[]
+>(
+  key: string = 'action',
+): ((arg: R) => Promise<R>) => {
+  const params = useQueryParams();
+  const actionParams = params.get(key);
+
+  const [actionTypes] = useMemo(() => {
+    let parsedAction;
+    try {
+      parsedAction = JSON.parse(actionParams);
+    } catch {
+      return [[]];
+    }
+    if (!parsedAction) return [[]];
+    if (Array.isArray(parsedAction)) {
+      return parsedAction.reduce(
+        ([types, payload], a) => {
+          types.push(a.type);
+          payload[a.type] = a.payload;
+          return [types, payload];
+        },
+        [[]],
+      );
+    }
+    return [[parsedAction.type], { [parsedAction.type]: parsedAction.payload }];
+  }, [actionParams]);
+
+  return useCallback(
+    async (arg: R) => {
+      if (actionTypes.includes(INCONTEXT_ACTIONS_CONNECTS_TO)) {
+        await doConnectsToBinding<R>(arg, INCONTEXT_ACTIONS_CONNECTS_TO);
```

Comment:
Is this `await` statement executed in context of async aware code?

Why not just start the async operation without `await` and be done with it, since it seems we're not interested in handling the resulting promise?

(I guess this part of the code was taken from the original code.)

Reply:
I think it's sometimes awaited and sometimes it isn't. I'd rather keep it here in case any consumer wants to await on this